### DISCUSSION TITLE
Thank you

### DISCUSSION THREAD

**Discussion by tensiondriven** [make-all/tuya-local#886] [General]
I've been struggling with a Pioneer Diamante WYT-series mini split / heat pump since I got it; the wifi connection works with Tuya but the HA Tuya integration doesn't translate the temperatures correctly, resulting in the gauge displaying values like "150 F".  

I switched over to tuya-local, chose one of the presets available, and it's working swell now.

Github issues / discussions are so often filled with problems; thought I'd take a sec to share how this repo provided a solution, with relative ease (once i got the device ids, etc).

Now, can tuya-local change my Zemismart roller blinds to work in increments of 1% instead of 10%?  That is the question.

**Comment by make-all** [comment]
Regarding the increments, it depends on the device capabilities itself.  I haven't looked in detail at this specific case, but normally any restrictions will be taken from the details at iot.tuya.com.  If they specify a range of 0-10 with a scale factor of 10, then almost certainly there is no possibility to use smaller increments.  If they specify 0-100 with a step size of 10, I would have followed that for the config, as we cannot predict the behaviour of the device if given in between input.  If you want to try for yourself, you can edit the step size in the config and report back on how it is working - but the device may be rounding/shifting that input internally, and it may be doing it in an unsafe way that causes overflow to cause side effects.

**Reply by make-all** [reply]
I looked at the config now, and there is no restriction in the config on the step size, so any restriction must be coming from either Home Assistant's UI or the device itself.  If the device itself, it might help to add the same step size in the config so that the UI snaps to the actual steps rather than allowing full control down to 1% intervals but the curtain only responding to changes > 10%.
